app/controllers/user/sessions_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
```

Comment:
How did all these devise controllers get generated?
Is it necessary to generate all these? 
What happens if I delete all these ` Devise::SessionsController` controllers

Reply:
There's a controller generator class in devise module where create_controllers method generates all the controllers, we can even specify which controllers we want to generate by passing them as an option in ``` rails generate devise:controllers``` command
```
def create_controllers
        @scope_prefix = scope.blank? ? '' : (scope.camelize + '::')
        controllers = options[:controllers] || CONTROLLERS
        controllers.each do |name|
          template "#{name}_controller.rb",
                   "app/controllers/#{scope}/#{name}_controller.rb"
        end
      end
``` 
If we delete Delete all these controllers then it will raise missing controller error